### PR TITLE
Add skeleton for Checkout Payment Trustly page

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Add skeleton for Payment page with Trustly.

## Justify why they are needed

So we can migrate away from Giraffe.

Excluded: API integration, Checkout integration.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2102** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2102" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2102?utm_source=stack-comment).

File: apps/store/src/components/CheckoutPaymentTrustlyPage/CheckoutPaymentTrustlyPage.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { Heading, Space, Text, theme } from 'ui'
+import { CheckoutStep } from '@/components/CheckoutHeader/Breadcrumbs'
+import { CheckoutHeader } from '@/components/CheckoutHeader/CheckoutHeader'
+import { GridLayout } from '@/components/GridLayout/GridLayout'
+import { TrustlyIframe } from '@/services/trustly/TrustlyIframe'
+import { RoutingLocale } from '@/utils/l10n/types'
+import { PageLink } from '@/utils/PageLink'
+import { SuccessState } from './SuccessState'
+
+const LOGGER = datadogLogs.createLogger('trustly')
```
Comment (alebedev) [inline apps/store/src/components/CheckoutPaymentTrustlyPage/CheckoutPaymentTrustlyPage.tsx:16]: Nit: `trustlyLogger`?

File: apps/store/src/pages/api/payment/adyen-callback/[shopSessionId]/[locale].ts [hunk 2]
```diff
@@ -22,7 +21,7 @@ export const handler = async (req: NextApiRequest, res: NextApiResponse) => {
     PageLink.checkoutPayment({
       locale,
       shopSessionId,
-      authStatus: AuthStatus.Success,
+      // authStatus: AuthStatus.Success,
```
Comment (alebedev) [inline apps/store/src/pages/api/payment/adyen-callback/[shopSessionId]/[locale].ts:24]: Commented code, what is intended state?

File: apps/store/src/components/CheckoutPaymentTrustlyPage/SuccessState.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import styled from '@emotion/styled'
+import { useTranslation } from 'next-i18next'
+import { CheckIcon, Text, theme } from 'ui'
+
+export const SuccessState = () => {
+  const { t } = useTranslation('checkout')
+
+  return (
+    <Centered>
+      <CheckIcon color={theme.colors.greenElement} />
```
Comment (alebedev) [inline apps/store/src/components/CheckoutPaymentTrustlyPage/SuccessState.tsx:10]: Nit: do we want to make it large?

File: apps/store/src/services/trustly/TrustlyIframe.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+type Props = {
+  url: string
+  onSuccess: () => void
+  onFail: () => void
+}
+
+export const TrustlyIframe = ({ url, onSuccess, onFail }: Props) => {
+  useEffect(() => {
+    const handler = (event: MessageEvent) => {
+      const trustlyOrigin = new URL(url).origin
```
Comment (alebedev) [inline apps/store/src/services/trustly/TrustlyIframe.tsx:17]: Nit: you can built it once, outside `handler`.  Not much of efficiency gain, but makes `hander` easier to read

File: apps/store/src/services/trustly/TrustlyIframe.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    return () => window.removeEventListener('message', handler)
+  }, [url])
+
+  const handleLoad: ReactEventHandler<HTMLIFrameElement> = (event) => {
+    const pathname = event.currentTarget.contentWindow?.location.pathname
+    if (pathname?.endsWith(SUCCESS_SUFFIX)) {
+      onSuccess()
+    }
+
+    if (pathname?.endsWith(FAIL_SUFFIX)) {
```
Comment (alebedev) [inline apps/store/src/services/trustly/TrustlyIframe.tsx:36]: Nit: `else if` would make control flow easier to read

**Review by alebedev — APPROVED** [review]
Good step, some nit comments inside

File: apps/store/src/components/CheckoutPaymentTrustlyPage/SuccessState.tsx (hunk 3 above)
Comment (robinandeer) [reply apps/store/src/components/CheckoutPaymentTrustlyPage/SuccessState.tsx:10]: hehe I just followed the design 🙃

File: apps/store/src/services/trustly/TrustlyIframe.tsx (hunk 4 above)
Comment (robinandeer) [reply apps/store/src/services/trustly/TrustlyIframe.tsx:17]: hmm but it depends on the URL which is passed as a prop... isn't it easy enough to define inline?

**Pull request merged**